fe/src/project/AspNetCore/shared/components/DownloadCenterFiles/DownloadCenterFiles.js

```diff
@@ -86,19 +101,47 @@ const DownloadCenterFiles = (props) => {
     const handleChangePage = (event, newPage) => {
         const startDisplayFiles = newPage * FilesConstants.defaultPageSize();
         setPage(newPage)
-        setFiles(props.files.slice(startDisplayFiles, startDisplayFiles + FilesConstants.defaultPageSize()))
+
+        let filesToSearch = props.files
+
+        if (searchTerm){
+            filesToSearch = searchedFiles;
+        }
+
+        setFiles(filesToSearch.slice(startDisplayFiles, startDisplayFiles + FilesConstants.defaultPageSize()))
+    }
+
+    const handleOnChange = (event) => {
+        setSearchTerm(event.target.value);
+    }
+
+    const handleSearchButton = () => {
+        const filesFromSearch = props.files.filter(x => x.name.toLowerCase().startsWith(searchTerm.toLowerCase()));
```

Comment:
**suggestion:** we need to perform search in the BE and return results to the FE. It won't work with the pagination because you will have only a subset of all files here.

Reply:
Hi @dawiddworak88 

Pagination is working, but you prefer be?

If true, i must add endpoints everywhere for files

Reply:
Yes, let's do it :)